backend/tests/Tests/Execution.Tests.fs

```diff
@@ -600,6 +616,68 @@ let testMatchPreview : Test =
       // TODO: constructor around a constructor around a value
       ]
 
+
+let testLetPreview : Test =
+  let letID = gid ()
+
+  let createLetPattern
+    (patternMatch : LetPattern)
+    (expr : Expr)
+    (finalExpr : Expr)
+    : Task<Map<id, AT.ExecutionResult>> =
+    task {
+      let ast = ELet(letID, patternMatch, expr, finalExpr)
+
+      let! results = execSaveDvals "let-preview" [] [] [] ast
+      let stuff = results |> Dictionary.toList |> Map
+
+      return stuff
+    }
+
+  testList
+    "test let preview"
+    [ testTask "let hello = 1 in hello" {
+        let lpID = gid ()
+
+        let letPattern = LPVariable(lpID, "hello")
+        let assignExpr = eInt 1
+        let retExpr = eVar "hello"
+
+        let! result = createLetPattern letPattern assignExpr retExpr
+
+        Expect.equal
+          (Map.get lpID result)
+          (Some(AT.ExecutedResult(DInt 1L)))
+          "hello assigned correctly"
+      }
+
+      testTask "let (x, y) = (1, 2) in y" {
```

Comment:
@pbiggar is this what you had in mind?

Reply:
This looks like a good start to me!

What happens upon `let (x, y) = (1, 1.0/0.0) in x`?

Without care, the interpreter would likely fully halt execution upon the `1/0` exception. I suppose the intent here is that the `x` variable is still visited/previewed.

Reply:
following up on a conversation had in a discord call:
my claim here was incorrect - the RHS of this expr will result in a Dval of an Error, so the "tuple-ness" will be lost, and `x` should not be expected to have the value of `1` even "in preview mode" or anything